zero_functional.nim

```diff
@@ -1243,16 +1258,20 @@ zf_inline exists(search: bool):
 
 ## Implementation of the 'find' command.
 ## Searches the input for a given expression. Returns an option value.
-zf_inline find(cond: bool):
+zfInline find(cond: bool):
   init:
-    result = none(it.type)
+    var initAccu = true
   loop:
-    if cond:
+    if initAccu:
+      # type of it must be assigned here because it may have been changed
+      result = none(it.type)
+      initAccu = false
+    elif cond:
       return some(it)
```

Comment:
why was this change necessary?

Reply:
This is a stopgap fix for it.type returning the original element type instead of the current one. It makes test "find with changed type" pass.

Reply:
OK - I wrote that comment before I see why the other tests were commented out. Same issue with the `init` section here I guess.
First I want to fix it more generally, then it should still be efficient... Finding out the output type is really tough - that's why this whole thing with the extra proc and using its return type... I tried a lot of stuff but a hit a lot of dead ends there. Maybe meanwhile it is easier in nim...?
If you have any ideas you're most welcome! But I think I have to think about it thoroughly.
You can get ideas of course if you look at the output of `->>` and try to come up with a general solution replacing the generated code by some other new code. Biggest problem when creating the macros is often that some information is retrieved later in the process of parsing - so finding out a return type before its use is tough.
Hope we can find a solution soon :-)

Reply:
btw - did you check this solution works also when finding the first (or only) element?